docker/jupyterhub_config.py

```diff
@@ -0,0 +1,79 @@
+import os
+import requests
+import socket
+
+from tornado import web
+
+from google.cloud import secretmanager_v1beta1 as secretmanager
+
+def access_secret_version(project_id, secret_id, version_id):
+  """
+  Accesses the payload for the given secret version if one exists. The version
+  can be a version number as a string (e.g. "5") or an alias (e.g. "latest").
+  """
+  client = secretmanager.SecretManagerServiceClient()
+  name = client.secret_version_path(project_id, secret_id, version_id)
+  response = client.access_secret_version(name)
+  payload = response.payload.data.decode('UTF-8')
+  return payload
+
+
+def is_true(boolstring: str):
+  """ Converts an environment variables to a Python boolean. """
+  if boolstring.lower() in ("true", "1"):
+    return True
+  return False
+
+# Listens on all interfaces.
+c.JupyterHub.hub_ip = '0.0.0.0'
+
+# Hostname that Cloud Dataproc can access to connect to the Hub.
+c.JupyterHub.hub_connect_ip = socket.gethostbyname(socket.gethostname())
+
+# Template for the user form.
+c.JupyterHub.template_paths = ['/etc/jupyterhub/templates']
+
+# Opens on JupyterLab instead of Jupyter's tree
+c.Spawner.default_url = os.environ.get('SPAWNER_DEFAULT_URL', '/lab')
+
+# The port that the spawned notebook listens on for the hub to connect
+c.Spawner.port = 12345
+
+print(os.environ)
+
+# Spawner
```

Comment:
This file now needs:

c.JupyterHub.proxy_class = 'redirect-proxy'

Reply:
Done.